src/main/java/org/zalando/nakadi/service/subscription/state/ClosingState.java

```diff
@@ -39,9 +43,10 @@ public void onExit() {
 
     @Override
     public void onEnter() {
-        final long timeToWaitMillis = getParameters().commitTimeoutMillis - (System.currentTimeMillis()
-                - lastCommitMillis);
-        if (timeToWaitMillis > 0) {
+        final long timeToWaitMillis = getParameters().commitTimeoutMillis -
```

Comment:
Well, it is exactly the cause of the bug. State travels thru following steps:
1. ctor
1. onEnter
1. onExit

The one that makes sense for us is `onEnter` because all the preconditions for this state should be calculated in that moment. That is why it is wrong to calculate something that dynamically changed on `ctor` stage.